extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/PythonTableDataService.java

```diff
@@ -162,11 +182,24 @@ public void getTableLocations(
                 @NotNull final TableDefinition definition,
                 @NotNull final TableKeyImpl tableKey,
                 @NotNull final Consumer<TableLocationKeyImpl> listener) {
+            final AsyncState<Boolean> asyncState = new AsyncState<>();
+
             final BiConsumer<TableLocationKeyImpl, ByteBuffer[]> convertingListener =
-                    (tableLocationKey, byteBuffers) -> processTableLocationKey(definition, tableKey, listener,
-                            tableLocationKey, byteBuffers);
+                    (tableLocationKey, byteBuffers) -> {
+                        try {
+                            processTableLocationKey(definition, tableKey, listener, tableLocationKey, byteBuffers);
+                        } catch (final RuntimeException e) {
+                            asyncState.setError(e);
```

Comment:
In a similar case for `table_schema`, you wrapped in an `IllegalArgumentException`.

Reply:
You're right; I would rather not wrap in table_schema -- that should catch a `RuntimeException` and not `Exception`.

Reply:
Well, technically the wrapper in table_schema specifies which schema failed to parse.. I'll clarify tomorrow.